lightning/src/ln/outbound_payment.rs

```diff
@@ -902,16 +902,63 @@ impl OutboundPayments {
 			}
 		}
 
-		let mut route = router.find_route_with_id(
-			&node_signer.get_node_id(Recipient::Node).unwrap(), &route_params,
+		let payer = node_signer.get_node_id(Recipient::Node).unwrap();
+		let route = match router.find_route_with_id(
+			&payer,&route_params,
 			Some(&first_hops.iter().collect::<Vec<_>>()), inflight_htlcs(),
 			payment_hash, payment_id,
-		).map_err(|_| {
-			log_error!(logger, "Failed to find route for payment with id {} and hash {}",
-				payment_id, payment_hash);
-			RetryableSendFailure::RouteNotFound
-		})?;
+		) {
+			Ok(res) => Some(res),
+			Err(error) => {
+				if error.err == "Cannot generate a route to ourselves" {
+					let payment_secret = match recipient_onion.payment_secret {
+						Some(secret) => secret,
+						None => PaymentSecret([0; 32])
+					};
+					let payment_preimage = PaymentPreimage([0; 32]);
```

Comment:
We should use `inbound_payment::get_payment_preimage` for this, not set it to a dummy value